tools/sync_voodoo_palette.py

```
@@ -0,0 +1,269 @@
+    """
+    with open(tokens_path, "rt") as f:
+        tokens = json.load(f)
+
+    try:
+        palette = tokens["colors"]["dark"]["content"]["palette"]
+    except KeyError:
+        raise SystemExit(
+            "Unexpected token shape in '%s': no colors.dark.content.palette. "
+            "The design system's token tree may have been restructured."
+            % tokens_path
+        )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)